tests/10apidoc/03events-initial.pl

```diff
@@ -126,9 +127,13 @@ sub flush_events_for
    });
 }
 
+# Note that semantics are undefined if calls are interleaved with differing
+# $room_ids for the same user.
+#
+# if $room_id is undefined, all joined rooms will be listened to.
 sub await_event_for
 {
-   my ( $user, $filter ) = @_;
+   my ( $user, $filter, $room_id ) = @_;
```

Comment:
It's quite nice to keep the filter sub last. I'd suggest

    my ( $user, @params ) = @_;
    my $filter = pop @params;